drivers/hmis/spec/models/hmis/ce/match_engine_spec.rb

```diff
@@ -215,12 +284,89 @@ def create_assessment_with_cde(client, cde_values)
       end
     end
   end
+
+  context 'when evaluating enrollment-based policies' do
+    include_context 'with enrolled test clients'
+
+    describe 'project-type-based filtering' do
+      describe 'when requiring open enrollment in a CE (14) project' do
+        # Requirement: must have open enrollment in Coordinated Entry (14) project type
+        let(:requirement_expression) { 'INCLUDES(open_enrollment_project_types, PROJECT_TYPE("CE"))' }
+
+        it 'includes clients with open enrollments at the correct project type' do
+          results = generate_candidates(pool, clients)
+          expected_clients = [client_enrolled_in_ce, client_wip_enrolled_in_ce]
+          expect(results.map(&:client_id).sort).to eq(expected_clients.map { |c| c.destination_client.id }.sort)
+        end
+
+        it 'excludes clients with exited enrollments at the correct project type' do
+          results = generate_candidates(pool, clients)
+          expect(results.map(&:client_id)).not_to include(client_exited_from_ce.destination_client.id)
+        end
+
+        it 'excludes clients with no enrollments' do
+          results = generate_candidates(pool, clients)
+          expect(results.map(&:client_id)).not_to include(client_unenrolled.destination_client.id)
+        end
+
+        it 'excludes clients that are only enrolled in projects of other types' do
+          results = generate_candidates(pool, clients)
+          expect(results.map(&:client_id)).not_to include(client_enrolled_in_es.destination_client.id)
+        end
+      end
+
+      describe 'when requiring open enrollment in a CE (14) project, excluding WIP enrollments' do
+        # Requirement: must have open enrollment in Coordinated Entry (14) project type, excluding WIP enrollments
+        let(:requirement_expression) { 'INCLUDES(open_enrollment_project_types_excluding_incomplete, PROJECT_TYPE("CE"))' }
+
+        it 'excludes clients with WIP (incomplete) enrollments at the project type' do
+          results = generate_candidates(pool, clients)
+          expect(results.map(&:client_id)).not_to include(client_wip_enrolled_in_ce.destination_client.id)
+        end
+
+        it 'includes clients with open enrollments at the correct project type' do
+          results = generate_candidates(pool, clients)
+          expect(results.map(&:client_id).sort).to eq([client_enrolled_in_ce.destination_client.id])
+        end
+      end
+
+      describe 'when requiring no open enrollments in PH project types' do
+        # Requirement: must NOT have open enrollment in any Permanent Housing project (3, 9, 10, 13)
+        let(:requirement_expression) { 'EXCLUDES(open_enrollment_project_types, PROJECT_TYPE("PH_PSH")) AND EXCLUDES(open_enrollment_project_types, PROJECT_TYPE("PH_PH")) AND EXCLUDES(open_enrollment_project_types, PROJECT_TYPE("PH_OPH")) AND EXCLUDES(open_enrollment_project_types, PROJECT_TYPE("PH_RRH"))' }
```

Comment:
this is a bit verbose but should work. Maybe in the future we might want something like an intersection method and an alias to all these project types. I guess we can see what the UI looks like.

A half-baked idea:
`NOT INTERSECTS(open_enrollment_project_types, PH_PROJECT_TYPES())`